viewing_party/party.py

```diff
@@ -1,23 +1,189 @@
+def get_most_watched_genre(user_data):
+    genre_freq_dict = {}
+    
+    for movie_dict in user_data["watched"]:
+
+        if movie_dict["genre"] not in genre_freq_dict:
+            genre_freq_dict[movie_dict["genre"]] = 1
+        else:
+            genre_freq_dict[movie_dict["genre"]] += 1
+    
+    if user_data["watched"] == []:
+        max_genre = None
+    else:
+        max_genre = max(genre_freq_dict, key=genre_freq_dict.get)
+
+    return max_genre
+  
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    #put each grouops watched movie titles in a set
+    my_watched_set = set()
+    friends_watched_set = set()
+
+    for movie_dict in user_data["watched"]:
+        my_watched_set.add(movie_dict["title"])
+    
+    for friends_list in user_data["friends"]:
+        for friends_watched_dict, friends_watched_list in friends_list.items():
+            for friends_movie_dict in friends_watched_list:
+                friends_watched_set.add(friends_movie_dict["title"])
+
+    
+#compare sets and output movies in watched that aren't in friends
+    user_watched_diffs_titles = my_watched_set.difference(friends_watched_set)
+    user_watched_dif_list = []
+    for movie_dict in user_data["watched"]:
+        if movie_dict["title"] in user_watched_diffs_titles:
+            user_watched_dif_list.append(movie_dict)
+  
+    return user_watched_dif_list
+
+def get_friends_unique_watched(user_data):
+    user_movies_list = []
+    for movie_info in user_data["watched"]:
+        user_movies_list.append(movie_info)
+    #make a list of friends movies
+    friends_movies_list = []
+    for indi_friends_dict in user_data["friends"]:
+        for key, friend_list in indi_friends_dict.items():
+            for movie_info in friend_list:
+                friends_movies_list.append(movie_info)
+    #if friend movie not in user movie, add to potential watchlist(unique_friend_movies)
+    user_hasnt_watched = [] #list of dictionaries with movie info friends have watched
+    potential_movies = []
+    
+    for movie_info in friends_movies_list:
+        if movie_info not in user_movies_list:
+            user_hasnt_watched.append(movie_info)
+            
+    user_hasnt_watched_dict = {}
+    for movie in user_hasnt_watched:
+        user_hasnt_watched_dict[movie["title"]] = movie
+    
+    for title, movie in user_hasnt_watched_dict.items():
+        potential_movies.append(movie)
+
+    return potential_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommended_movies_list = []
+#1. loop through the friends movies and see which ones the
+#user hasn't watched.

```

Comment:
This feels like what `get_friends_uniqued_watched` does and it looks like this code is a repeat of that function. Since this work already exists elsewhere, you should call that method here. A potential solution:
```
unique_friend_movies = get_friends_unique_watched(user_data)
recs = []
```
and then lines 156 - 160 and that's it!